PortalSwift/Sources/PortalSwift/Utils/PortalRequests.swift

```diff
@@ -0,0 +1,177 @@
+
+  public func post(_ from: URL, withBearerToken: String? = nil, andPayload: Encodable? = nil) async throws -> Data {
+    var request = URLRequest(url: from)
+
+    // Add required request headers
+    if let token = withBearerToken {
+      request.addValue("Bearer \(token)", forHTTPHeaderField: "Authorization")
+    }
+
+    request.addValue("application/json", forHTTPHeaderField: "Accept")
+    request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+
+    // Make sure we're sending a POST request with the data provided
+    request.httpMethod = "POST"
+
+    if let payload = andPayload {
+      request.httpBody = try JSONEncoder().encode(payload)
+      request.addValue("\((String(data: request.httpBody!, encoding: .utf8) ?? "").count)", forHTTPHeaderField: "Content-Length")
+    }
+
+    // Send the request
+    let (data, response) = try await getURLSession().data(for: request)
+
+    // Check the reponse status
+    guard let httpResponse = response as? HTTPURLResponse else {
+      throw PortalRequestsError.couldNotParseHttpResponse
+    }
+    guard httpResponse.statusCode < 300 else {
+      throw self.buildError(httpResponse, withData: data)
+    }
+
+    return data
+  }
+
+  public func postMultiPartData(
+    _ from: URL,
+    withBearerToken: String,
+    andPayload: String,
+    usingBoundary: String
+  ) async throws -> Data {
+    var request = URLRequest(url: from)
+
+    // Add required request headers
+    request.addValue("application/json", forHTTPHeaderField: "Accept")
+    request.addValue("Bearer \(withBearerToken)", forHTTPHeaderField: "Authorization")
+    request.addValue("multipart/related; boundary=\(usingBoundary)", forHTTPHeaderField: "Content-Type")
+
+    request.httpMethod = "POST"
+    request.httpBody = andPayload.data(using: .utf8)
+
+    let (data, response) = try await getURLSession().data(for: request)
+
+    // Check the reponse status
+    guard let httpResponse = response as? HTTPURLResponse else {
+      throw PortalRequestsError.couldNotParseHttpResponse
+    }
+    guard httpResponse.statusCode < 300 else {
+      throw self.buildError(httpResponse, withData: data)
+    }
+
+    return data
+  }
+
+  private func buildError(_ response: HTTPURLResponse, withData: Data) -> Error {
+    let statusText = String(data: withData, encoding: .utf8) ?? ""
+    if response.statusCode < 400 {
+      return PortalRequestsError.redirectError("\(response.statusCode) - \(statusText)")
+    } else if response.statusCode < 500 {
+      return PortalRequestsError.clientError("\(response.statusCode) - \(statusText)")
+    } else {
+      return PortalRequestsError.internalServerError("\(response.statusCode) - \(statusText)")
+    }
+  }
+
+  private func getURLSession() -> URLSession {
+    var configuration = URLSessionConfiguration.default
+    configuration.requestCachePolicy = .reloadIgnoringLocalCacheData
+    configuration.urlCache = URLCache(memoryCapacity: 0, diskCapacity: 0, diskPath: nil)
```

Comment:
Amazing 🙌  This should resolve a lot of the confusion where customers were getting cached results previously